src/firewall/core/ipXtables.py

```diff
@@ -684,7 +684,14 @@ def build_set_policy_rules(self, policy):
             if table == "nat":
                 continue
             for chain in BUILT_IN_CHAINS[table]:
-                rules.append(["-t", table, "-P", chain, _policy])
+                if table == "filter" and policy == "DROP":
+                    p = drop_details[chain]
+                    if p == "REJECT":
+                        rules.append(["-t", table, "-A", chain, "-j", "REJECT"])
+                        p = "DROP"
+                else:
+                    p = _policy
+                rules.append(["-t", table, "-P", chain, p])
```

Comment:
In general, I think we should always pass valid `drop_details`. This allows us to make simpler decisions down the chain, e.g. this section could be rewritten as:

```python
                if table == "filter" and policy == "DROP" \
                   and drop_details[chain] == "REJECT":
                    rules.append(["-t", table, "-A", chain, "-j", "REJECT"])
                rules.append(["-t", table, "-P", chain, drop_details[chain]])
```

Reply:
I don't think it's simpler. It extends the purpose of the variable, making it harder to understand.

`drop_details` is simple. If the `policy` is `"DROP"`, then these are the details (the settings from `ReloadPolicy=`).

If the argument also applies to `ACCEPT`, what does it mean there?

Reply:
So rename it to `policy_details`. It's then generic in all cases and we don't need an special casing.

Reply:
with `policy="PANIC"|"ACCEPT"`, what are callers supposed to set the `policy_details` to?

Reply:
For PANIC it's all "DROP". For accept is all "ACCEPT".